packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -1076,11 +1076,16 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
 
 		// Watch the minimum sequence number and be ready to update as needed
 		if (this.minSequenceNumber > message.minimumSequenceNumber) {
-			// pre-0.58 error message: msnMovesBackwards
-			throw new DataCorruptionError(
-				"Found a lower minimumSequenceNumber (msn) than previously recorded",
+			// This happens when something changes out from under the file on the server,
+			// e.g. Database rollback or storage otherwise being modified outside of Fluid's purview.
+			// This is a fatal error with data loss for this client, but others will likely be able to connect and continue.
+			throw DataProcessingError.create(
```

Comment:
@vladsud you had this comment somewhere in the conversation:

> If we believe that all these errors are related only to DB rollback (and not to bugs in PUSH), then we should likely propagate them better to the user. I.e. instead of us saying that content changed or MSN are out of whack, we should throw an error that says "document content was overwritten / restored to previous state" and ensure that Loop (and other apps) properly propagate that kind of message to the user - this gives us a bit better user experience (but same result - data loss).

I also considered using Driver Error Type "fileOverwrittenInStorage".  Any thoughts on which you'd prefer between that and "DataProcessingError"?

Reply:
Suggestions welcome for wording the error message and comment too, btw.

Reply:
i prefer DataProcessingError for the reasons in my other comment.

Reply:
Yeah thanks for that input, makes a lot of sense.  I'll take another look on Monday at putting something in the driver layer.

Reply:
i think it is fine to keep this check, and have a check in the driver. the driver check will supersede this check as the driver will see all data before the deltamanager. this then acts as the default check, but lets drivers do more specific errors where more can be known.

Reply:
I looked at adding a check in the Driver.  I don't think it's easy.  The DeltaManager does all the work to collect all the ops in perfect sequential order, leveraging the driver to stream/fetch ops to build up the inbound queue. Only after their queued can we reason over invariants like this.

So it seems like if we want the driver to be able to interpret this case we would instead need to add some kind of hook here.  Out of scope for this change.

Lmk if I'm missing something, if there is a quick way to instrument this in the driver I can do a follow-up PR.